lib/Finance/Quote/YahooJapan.pm

```diff
@@ -175,30 +175,27 @@ sub _convert_quote {
 sub _scrape {
     my $tree = shift;
     my %quotes = ();
-
-    my $container = $tree->look_down('id', 'sr');
-    if (defined $container) {
-        for my $e ($container->look_down('class', '_239Zl3PI')) {
-            my $sym = $e->look_down('class', 'CGplzQf_')->as_text;
-            my ($date, $time) = _parse_datetime($e->look_down('class', '_2JynCBOQ')->as_text);
-            my $quote = {
-                name  => $e->look_down('class', '_2MnVoYg5')->as_text,
-                price => $e->look_down('class', '_3rXWJKZF')->as_text,
-                date  => $date,
-                time  => $time
-            };
-            $quote->{'price'} =~ tr/.0-9//cd;   # strip commas, etc.
-
-            # for a stock code, register a duplicate quote with market letter
-            if ($sym =~ /^[0-9A-Z]{2}[0-9][0-9A-Z][0-9]?$/) {
-                my $pat = qr/(?:quote\/|code=)($sym\.[A-Z])/;
-                $e->look_down('_tag', 'a', 'href', $pat)->attr('href') =~ $pat;
-                $quotes{lc $1} = $quote if (defined $1);
-            }
-
-            # XXX destructive when a stock quote from other market already exists
-            $quotes{$sym} = $quote;
+    my @nodes =  $tree->findnodes('//*[@id="sr"]/div/ul/li/article');
```

Comment:
Is the xpath necessary? Just
`$container->find('article')`
should work well.

Reply:
I thouhgt that removing the following three lines and using the xpath would be more readable.
I leave it to the maintainer to decide which is better.

```perl
    my $container = $tree->look_down('id', 'sr');
    if (defined $container) {
...
   }
```